observer.go

```diff
@@ -0,0 +1,15 @@
+package opentracing
+
+// Observer can be registered with the Tracer to recieve notifications
```

Comment:
@yurishkuro I have added your suggestions and made the comments a bit more generic.
@bhs I have added a "Note" about the `Observer` API being in alpha.